apps/website/src/pages/insights/epics/[epic].tsx

```diff
@@ -1,82 +1,306 @@
+    {
+      epicName: epicName as string,
+      from: initialDates.from,
+      to: initialDates.to,
+    },
+    {
+      initialData: props.burnup,
     }
+  )
+
+  const { data: dataCounter } = useGetEpicIssuesCountQuery({
+    where: {
+      epic_name: { _eq: epicName as string },
+      ...(selectedAuthors.length > 0 && {
+        author: { _in: selectedAuthors },
+      }),
+      ...(selectedAssignees.length > 0 && {
+        assignee: { _in: selectedAssignees },
+      }),
+      ...(selectedRepos.length > 0 && {
+        repository: { _in: selectedRepos },
+      }),
+      title: { _ilike: `%${debouncedSearchFilter}%` },
+    },
+  })
+
+  const count = {
+    total: dataCounter?.gh_epic_issues.length,
+    closed: dataCounter?.gh_epic_issues.filter(issue => issue.closed_at).length,
+    open: dataCounter?.gh_epic_issues.filter(issue => !issue.closed_at).length,
   }
 
-  return {
-    props: {
-      epic,
-      breadcrumbs: [
-        {
-          label: 'Epics',
-          href: '/insights/epics',
-        },
-        {
-          label: epic.title,
-          href: `/insights/epics/${epic.id}`,
-        },
+  const burnup = dataBurnup?.gh_burnup || []
+
+  const { data, isFetching, fetchNextPage, hasNextPage, isFetchingNextPage } =
+    useInfiniteQuery(
+      [
+        'getIssuesByEpic',
+        epicName,
+        activeTab,
+        selectedAssignees,
+        selectedRepos,
+        selectedAuthors,
+        orderByValue,
+        debouncedSearchFilter,
       ],
+      async ({ pageParam = 0 }) => {
+        const result = await api<
+          GetIssuesByEpicQuery,
+          GetIssuesByEpicQueryVariables
+        >(GET_ISSUES_BY_EPIC, {
+          where: {
+            epic_name: { _eq: epicName as string },
+            stage: { _eq: activeTab },
+            ...(selectedAuthors.length > 0 && {
+              author: { _in: selectedAuthors },
+            }),
+            ...(selectedAssignees.length > 0 && {
+              assignee: { _in: selectedAssignees },
+            }),
+            ...(selectedRepos.length > 0 && {
+              repository: { _in: selectedRepos },
+            }),
+            title: { _ilike: `%${debouncedSearchFilter}%` },
+          },
+          limit: LIMIT,
+          offset: pageParam,
+          orderBy: orderByValue,
+        })
+
+        return result?.gh_epic_issues || []
+      },
+      {
+        getNextPageParam: (lastPage, pages) => {
+          if (lastPage.length < LIMIT) {
+            return undefined
+          }
+
+          return pages.length * LIMIT
+        },
+      }
+    )
+
+  const { data: filters } = useGetFiltersWithEpicQuery(
+    {
+      epicName: epicName as string,
     },
-  }
-}
+    {
+      initialData: props.filters,
+    }
+  )
 
-type Props = {
-  epic: Epic
-  breadcrumbs: BreadcrumbsProps['items']
-}
+  const issues =
+    data?.pages.reduce((acc, page) => {
+      return [...acc, ...page]
+    }, []) || []
```

Comment:
[flatMap](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) exists exactly for this use-case.
```ts
const issues = data?.pages.flatMap(page => page) || [];
```